apps/client/src/module/auth/reset-password-form.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+import { zodResolver } from "@hookform/resolvers/zod";
+import {
+    Button,
+    ControlledPasswordInput,
+} from "@src/module/common";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import { AuthFormContainer } from "./auth-form-container";
+import { resetPassword, ResetPasswordPayload } from "./auth-service";
+import { AuthRoute } from "./route";
+import { UserHelpMsgs } from "@src/utils/user-help-messgage";
+import { useState, useEffect } from "react";
+import { PasswordErrorMsgs, passwordPatten, ConfirmPasswordErrorMsgs } from "@src/utils/validation-message";
+
+
+type FormSchema = z.infer<typeof formSchema>;
+const formSchema = z
+    .object({
+        password: z.string()
+        .min(1, { message: PasswordErrorMsgs.Required })
+        .min(8, { message: PasswordErrorMsgs.MinLength })
+        .max(32, { message: PasswordErrorMsgs.MaxLength })
+        .regex(passwordPatten, { message: PasswordErrorMsgs.Invalid }),
+        confirmPassword: z.string()
+        .min(1, { message: ConfirmPasswordErrorMsgs.Required })
+        .min(8, { message: PasswordErrorMsgs.MinLength })
+        .max(32, { message: PasswordErrorMsgs.MaxLength })
+        .regex(passwordPatten, { message: PasswordErrorMsgs.Invalid }),
+    })
+    .refine((data) => data.password === data.confirmPassword, {
+        message: ConfirmPasswordErrorMsgs.Mismatch,
+        path: ["confirmPassword"],
+    });
+
+export function ResetPasswordForm() {
+    const [responseMessage, setResponseMessage] = useState('');
+    const [token, setToken] = useState('');
+
+    useEffect(() => {
+        const urlParams = new URLSearchParams(window.location.search);
```

Comment:
there's a different way to get url Params. You probably want to check the docs on NextJS.

Reply:
https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional